server/handler/base.go

```diff
@@ -146,32 +146,43 @@ func (b *Base) Evaluate(ctx context.Context, installationID int64, trigger commo
 
 	fetchedConfig, err := b.ConfigFetcher.ConfigForPR(ctx, prctx, client)
 	if err != nil {
-		return errors.WithMessage(err, fmt.Sprintf("failed to fetch policy: %s", fetchedConfig))
+		return errors.WithMessage(err, fmt.Sprintf("Failed to fetch configuration: %s", fetchedConfig))
 	}
 
-	return b.EvaluateFetchedConfig(ctx, prctx, client, fetchedConfig, trigger)
+	evaluator, err := b.ValidateFetchedConfig(ctx, prctx, client, fetchedConfig, trigger)
+
+	if evaluator == nil {
+		return err
+	}
```

Comment:
Note: Logic for generating `err` is now slightly different. This should not affect functionality. The changes are:
- When the config is **missing**: `nil` used to be returned, now an error with the config description (missing owner/repo/ref) is returned. 
  - A status API call is not made, so the error should be caught by the caller. I believe this will be logged.
- When the config is **invalid**: The config error is now wrapped with the config description (invalid owner/repo/ref).
- When the **evaluation** throws an error: That error is now wrapped with a message mentioning the policy could not be parsed.
These changes were made to simplify the logic in the details page generation.

Reply:
> When the config is missing: nil used to be returned, now an error with the config description (missing owner/repo/ref) is returned.

I think we'll want to either leave this as a `nil` error or mark it in some way that can be ignored by things other than the detail view. If I'm following the code correctly, while this won't result in a failed status, it will still appear as an unexpected error in logs and metrics.

> When the evaluation throws an error: That error is now wrapped with a message mentioning the policy could not be parsed.

I don't see this behavior in the implementation, but I did want to mention that evaluation errors can only happen after successful parsing.

Reply:
Ok, sounds good. Changed it so that missing configs no longer return an error. The details view will now generate it's own error when `evaluator` is `nil`, whereas base/issue_comment callers don't care (since they just skip missing configs).

Sorry, I meant to say when `ParsePolicy` is called to get the evaluator and an error is produced we now wrap that error with a [message](https://github.com/palantir/policy-bot/pull/265/files#diff-24d63556b451fd44ed18a6ea7f625e95bc19c861c7988f1f81a6272fc18d0eb7R200) mentioning the policy could not be parsed.